src/test/java/com/google/devtools/build/lib/authandtls/NetrcParserTest.java

```diff
@@ -0,0 +1,235 @@
+    String content =
+        "machine " + fooMachine + " login " + fooCredential.login() + " password " + fooCredential
+            .password() + "\n"
+            + "macdef init\n"
+            + "\tcd /pub\n"
+            + "\tmget *\n"
+            + "\tquit\n"
+            + "\n"
+            + "machine " + barMachine + " login " + barCredential.login() + " password "
+            + barCredential.password();
+    InputStream inputStream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8));
+
+    Netrc netrc = NetrcParser.parseAndClose(inputStream);
+
+    ImmutableMap<String, Credential> expectedCredentials = ImmutableMap.of(
+        fooMachine,
+        fooCredential,
+        barMachine,
+        barCredential
+    );
+    assertThat(netrc.getDefaultCredential()).isNull();
+    assertThat(netrc.getCredentials()).isEqualTo(expectedCredentials);
+    assertThat(netrc.getCredential(fooMachine)).isEqualTo(fooCredential);
+    assertThat(netrc.getCredential(barMachine)).isEqualTo(barCredential);
+  }
+
+  @Test
+  public void shouldOverrideDuplicatedMachineFields() throws IOException {
+    String content =
+        "machine " + fooMachine + " login overridden_user login " + fooCredential.login()
+            + " password overridden_pass password " + fooCredential
+            .password();
+    InputStream inputStream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8));
+
+    Netrc netrc = NetrcParser.parseAndClose(inputStream);
+
+    ImmutableMap<String, Credential> expectedCredentials = ImmutableMap.of(
+        fooMachine,
+        fooCredential
+    );
+    assertThat(netrc.getDefaultCredential()).isNull();
+    assertThat(netrc.getCredentials()).isEqualTo(expectedCredentials);
+    assertThat(netrc.getCredential(fooMachine)).isEqualTo(fooCredential);
+  }
+
+  @Test
+  public void shouldIgnoreMachinesAfterDefault() throws IOException {
+    String content =
+        "machine " + fooMachine + " login " + fooCredential.login() + " password " + fooCredential
+            .password() + "\n"
+            + "default login " + defaultCredential.login() + " password " + defaultCredential
+            .password() + "\n"
+            + "machine " + barMachine + " login " + barCredential.login() + " password "
+            + barCredential.password();
+    InputStream inputStream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8));
+
+    Netrc netrc = NetrcParser.parseAndClose(inputStream);
+
+    ImmutableMap<String, Credential> expectedCredentials = ImmutableMap.of(
+        fooMachine,
+        fooCredential
+    );
+    assertThat(netrc.getDefaultCredential()).isEqualTo(defaultCredential);
+    assertThat(netrc.getCredentials()).isEqualTo(expectedCredentials);
+    assertThat(netrc.getCredential(fooMachine)).isEqualTo(fooCredential);
+  }
+
+  @Test
+  public void shouldFailWithBadStartingContent() {
+    String content = "this is not netrc syntax";
```

Comment:
What do other tools do for bad .netrc files? Do they error out or do they silently ignore it?

Reply:
curl just silently ignore it. Should we also ignore the bad .netrc files silently?

Reply:
We've ran into issues python `requests` module ignores the error silently as well, not my favorite behavior, but I think it's fine to silently ignore here to keep it consistent.

Although just to be ideal, if there's a flag that can change the behavior, that'll be even better. (not blocking here)

Reply:
Would be good to respect `#`-style comments, as they're pretty common in netrcs.

Otherwise we should probably do whatever the starlark netrc parser does.

Reply:
> but I think it's fine to silently ignore here to keep it consistent.

Done.

> Would be good to respect #-style comments, as they're pretty common in netrcs.

Done.